File: problems/0_duplicate_character/tylertSolution.py
# this program both computes the answer for the input file and checks if it is correct
# to run the program run the command `python tylertSolution.py` in the terminal while in the directory

filename = "input.txt"
inputFile = open(filename, "r")

filename = "output.txt"
outputFile = open(filename, "r")

def hasDuplicate(str):
  seen = set()
  for c in str:
    if c in seen:
      return True
    else:
      seen.add(c)
  return False

def isCorrect(result):
  output = []
  for line in outputFile:
    output.append(line)

  for i in range(len(result)):
    str1 = str(result[i]).lower()
    str2 = output[i].lower().strip()
    if not str1 == str2:
      return False
  return True

result = []
for line in inputFile:
  res = hasDuplicate(line)
  result.append(res)

res = isCorrect(result)
if res:
  print("Correct")
else:
  print("Inncorrect")

File: problems/0_duplicate_character/tylertSolution.java
// to compile this code run `javac tylertSolution.java` in your terminal in this folder
// to run this code run `java tylertSolution input.txt > tyler.out`, this input.txt is a argument that is used to access the input file
// and the output is redirected to a file called tyler.out
// run `cat tyler.out` in your terminal to see the output


import java.io.FileNotFoundException;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.List;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.stream.Collectors;

public class tylertSolution {
  public static List<String> readFile(String filename) {
    List<String> lines = new ArrayList<String>();
    try {
      lines = Files.lines(Paths.get(filename)).collect(Collectors.toList());
    } catch (FileNotFoundException e) {
      System.out.println("File not found");
    } catch (IOException e) {
      System.out.println("IO Error");
    }
    return lines;
  }

  public static Boolean hasDuplicate(String line) {
    HashSet seen = new HashSet<Character>();
    for (int i = 0; i < line.length(); i++) {
      Character c = line.charAt(i);
      if (seen.contains(c)) {
        return true;
      } else {
        seen.add(c);
      }
    }
    return false;
  }

  public static void main(String[] args) {
    String filename = args[0];
    List<String> lines = readFile(filename);
    for (String line : lines) {
      System.out.println(hasDuplicate(line));
    }
  }
}